### DISCUSSION TITLE
How to handle two solar inputs?

### DISCUSSION THREAD

**Discussion by bensebborn** [slipx06/sunsynk-power-flow-card#297] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We have a Sunsynk with battery and solar which shows fine 

however we also have some additional solar (Solis) that isn’t connected to the Sunsynk. As such on other monitoring apps it’s just seen as reducing the house load before the Sunsynk sees it

eg

House load: 2kW
Solis Solar: 1kW
Net house load: 1kW

Sunsynk battery: 1kW
Grid: 0

I’ve setup the card to show the Solar correctly on the Sunsynk but I think if I add the Solis Solar to the sooar entities the calculations are wrong. 

So is it best to add the Solar as a house or non essential load instead?

thanks. 

### Expected behaviour

Additional solar can be shown corrextly

### Mode

Both

### Context / Reason

Being able to show solar directly connected to main inverter plus additional solar from existing inverter.

**Comment by bensebborn** [comment]
Here’s a comparison between our MyEnergi setup (icon at 5pm is the Sunsynk, icon at 4pm is Solis)

**Comment by bensebborn** [comment]
![IMG_5321](https://github.com/slipx06/sunsynk-power-flow-card/assets/160988/14fd468e-bd8c-4afc-a865-f43897697227)

**Comment by bensebborn** [comment]
![IMG_5322](https://github.com/slipx06/sunsynk-power-flow-card/assets/160988/23912a65-b392-406a-a28a-f221cd828783)

**Comment by slipx06** [comment]
You can try add it as an AUX input and let the card calculate the essential power (i.e don't provide an `essential_power` sensor).  It will use the formula 

essential = inverter_power_175 + grid_power_169 - aux_power_166
